enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+
+Similar to any other capability, the ingress capability:
+- can be enabled not only as part of the capability sets (`vCurrent`, `v4,14`) but explicitly by using [the additional enabled capability field](https://github.com/openshift/api/blob/3778e7a4a55241c36e66e43cb5d124f44938c094/config/v1/types_cluster_version.go#L433-L438).
+- if enabled, should appear in [the list of the enabled capabilities](https://github.com/openshift/api/blob/3778e7a4a55241c36e66e43cb5d124f44938c094/config/v1/types_cluster_version.go#L445-L448) of `ClusterVersion` object.
+- if enabled, cannot be disabled post install.
+
+#### HyperShift
+
+HyperShift doesn't use the OpenShift installer to spawn the user clusters. HyperShift deploys the cluster version operator to manage the cluster operators though.
+However not all the cluster operators are of use for HyperShift.
+The functionality of some cluster operators is fully replaced by the logic in the hosted control plane (e.g. authentication operator).
+The ingress operator is an example of a cluster operator whose life cycle is shared between the cluster version operator and HyperShift.
+HyperShift doesn't pass all the ingress operator's payload to the cluster version operator.
+Certain manifests are filtered out, most importantly: [the ingress operator's deployment](https://github.com/openshift/hypershift/blob/0a5ec529ce7ae6f238ed27be6eb5757cf70c41f9/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L75).
+The reason for this is that the ingress operator is considered as a part of the hosted control plane hence deployed on the control plane cluster.
+However [the ingress operator's APIs](#ingress-operator-apis) and some RBACs to manage those APIs are still kept in the cluster version operator's payload.
+The ingress capability comes handy here as HyperShift can leverage it to control the remaining (not filtered) payload.
+
+The HyperShift's API which will handle the hosted cluster capabilities is out of the scope of this enhancement. However this enhancement is built with the assumption that the `ClusterVersion`'s capability API (spec and status) is sufficient for any capability implementation in HyperShift.
+
+#### Standalone OpenShift
+
+Making the ingress operator optional for the standalone OpenShift is a bigger scope than for HyperShift.
+The changes impact the components which are not used (or used differently) in HyperShift like the installer, the authentication and the monitoring operators.
+To prevent the scope creep for [RFE-3395](https://issues.redhat.com/browse/RFE-3395), the following approach is proposed:
+- the cluster version operator should be able to add required capabilities.
```

Comment:
You can use italics to hint that you are introducing a new term that the reader isn't expected to know already.  
```suggestion
- The cluster version operator should be able to add _required capabilities_.
```
The phrase "required capabilities" makes me think that CVO will complain if the capability is not enabled.  However, if I correctly understand your description of the concept, the idea is that these "required capabilities" are enabled by CVO whether or not they are specified in the install-config.  Maybe "force-enabled capabilities" and `--force-enable=ingress` as the flag would be clearer?
```suggestion
- The cluster version operator should be able to add _force-enabled capabilities_.
```

Reply:
i like the terminology clarification

Reply:
`--force-enabled` feels aggressive, like "do this even if customer doesn't want this".  But the reason we're setting this is because the standalone customer does want this (because they want a functional cluster) even if they don't ask for `Ingress` explicitly.  Maybe we could use `--additional-implicitly-enabled-capabilities` or something?  It doesn't have to be short, because it's just going into a manifest, and not something anyone ever needs to type on the command line.

Reply:
I don't have a strong opinion on the naming of the flag. I'm still fine with `--required-capabilities`. Who has the final word in CVO repository?

Reply:
OTA team, of which @wking is a member, so i'm happy to let him pick whatever he's happy with.  This isn't a user-facing flag.

Reply:
Btw, the naming "required" is already used in the code of the cluster version operator for [the required feature sets](https://github.com/openshift/cluster-version-operator/blob/8913d38/pkg/cvo/cvo.go#L158-L160).

Reply:
`--always-enable-capabilities` is the name we agreed on. Closing the conversation.